hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/oauth2/ClientCredsTokenProvider.java

```diff
@@ -38,19 +42,27 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(AccessTokenProvider.class);
 
-
   public ClientCredsTokenProvider(final String authEndpoint,
-                                  final String clientId, final String clientSecret) {
-
-    Preconditions.checkNotNull(authEndpoint, "authEndpoint");
-    Preconditions.checkNotNull(clientId, "clientId");
-    Preconditions.checkNotNull(clientSecret, "clientSecret");
+      final String clientId, final String clientSecret)
+      throws AzureBlobFileSystemException {
+    validateClientCredsTokenProvider(
+        ConfigurationKeys.FS_AZURE_ACCOUNT_OAUTH_CLIENT_ENDPOINT, authEndpoint);
+    validateClientCredsTokenProvider(
+        ConfigurationKeys.FS_AZURE_ACCOUNT_OAUTH_CLIENT_ID, clientId);
+    validateClientCredsTokenProvider(
+        ConfigurationKeys.FS_AZURE_ACCOUNT_OAUTH_CLIENT_SECRET, clientSecret);
 
     this.authEndpoint = authEndpoint;
     this.clientId = clientId;
     this.clientSecret = clientSecret;
   }
 
+  private void validateClientCredsTokenProvider(final String configName,
```

Comment:
IMHO, it is better to move it to AbfsConfiguration.java. It will allow you to cover other token providers there as well.

Reply:
move this to the code which gets the config and instantiates the tokenprovider